src/engraving/dom/tie.cpp

```diff
@@ -355,30 +355,68 @@ void Tie::calculateDirection()
     }
 }
 
-void Tie::calculateIsInside()
+void Tie::calculateIsInside(bool isUp)
 {
     if (_tiePlacement != TiePlacement::AUTO) {
         setIsInside(_tiePlacement == TiePlacement::INSIDE);
         return;
     }
 
     const Note* startN = startNote();
+    const Note* endN   = endNote();
     const Chord* startChord = startN ? startN->chord() : nullptr;
-    const Note* endN = endNote();
-    const Chord* endChord = endN ? endN->chord() : nullptr;
+    const Chord* endChord   = endN ? endN->chord() : nullptr;
 
     if (!startChord || !endChord) {
         setIsInside(false);
         return;
     }
 
-    const bool startIsSingleNote = startChord->notes().size() <= 1;
-    const bool endIsSingleNote = endChord->notes().size() <= 1;
+    std::vector<Note*> startNotes = startChord->notes();
+    std::vector<Note*> endNotes   = endChord->notes();
+    size_t n_start = startNotes.size();
+    size_t n_end   = endNotes.size();
 
-    if (startIsSingleNote && endIsSingleNote) {
+    if (n_start <= 1 && n_end <= 1) {
         setIsInside(style().styleV(Sid::tiePlacementSingleNote).value<TiePlacement>() == TiePlacement::INSIDE);
     } else {
-        setIsInside(style().styleV(Sid::tiePlacementChord).value<TiePlacement>() == TiePlacement::INSIDE);
+        if (style().styleV(Sid::tiePlacementChord).value<TiePlacement>() == TiePlacement::AUTO) {
+            int startNotesAbove = 0, startNotesBelow = 0;
+            int endNotesAbove   = 0, endNotesBelow   = 0;
+            for (size_t i = 0; i < n_start; ++i) {
```

Comment:
```
for (Note* note : startNotes) {
    ...
}
```
that way you don't need n_start and you don't need to do startNotes[i] all the time.